pkg/miscellaneous/miscellaneous.go

```diff
@@ -61,6 +61,42 @@ func GetLatestFullSnapshotAndDeltaSnapList(store brtypes.SnapStore) (*brtypes.Sn
 	return fullSnapshot, deltaSnapList, nil
 }
 
+// Backup contains a full snapshot and all of its delta snapshots.
+type Backup struct {
+	FullSnapshot      *brtypes.Snapshot
+	DeltaSnapshotList brtypes.SnapList
```

Comment:
Is the distinction between full and delta snapshots important while copying? Could we not simply have a list of snapshots? We could still group them together in the `Backup` structure based on a full snapshot and its following delta snapshots. But by grouping them together in one list might simplify the rest of the copier logic. WDYT?

Reply:
I did a bit of refactoring based on your feedback and indeed it looks better that way. Functionally I don't find find differences.

Reply:
Yes, I didn't mean to have any change in functionality. Just to simplify the logic. That is all. Thanks for taking up the changes.